core/src/data_sources/data_source.rs

```diff
@@ -1266,6 +1166,130 @@ impl DataSource {
         Ok(documents)
     }
 
+    async fn retrieve_chunks_without_query(
+        &self,
+        store: Box<dyn Store + Sync + Send>,
+        qdrant_client: &Arc<QdrantClient>,
+        top_k: usize,
+        filter: &Option<SearchFilter>,
+    ) -> Result<Vec<(String, Chunk)>> {
+        let store = store.clone();
+
+        let (doc_ids, _) = store
+            .find_data_source_document_ids(
+                &self.project,
+                self.data_source_id(),
+                filter,
+                // with top_k documents, we should be guaranteed to have at
+                // least top_k chunks, if we make the assumption that each
+                // document has at least one chunk.
+                // TODO @fontanierh: confirm this is actually true
+                Some((top_k, 0)),
```

Comment:
@spolu  This is one assumption that may be untrue. If it is wrong, then in some unlucky cases we may end up with less chunks than top_k. 
But I don't really see how it's possible

Reply:
If "it works" even if we get some empty documents then it's fine :+1: can only be a comment to bring the reader attention to that.

Reply:
it does work 👍